src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -399,6 +399,14 @@ struct
     node.breadcrumb
 
   let clear_paths t = Hashtbl.clear t.table
+
+  module Ledger_transfer = Ledger_transfer.Make (Ledger) (Ledger.Db)
+
+  let rebuild t ledger state_hash =
+    Ledger_transfer.transfer_accounts ~src:ledger ~dest:t.root_snarked_ledger
```

Comment:
For this to be correct `Ledger_transfer` needs to empty the dest ledger. Should we just turn `root_snarked_ledger` mutable and make a new one here?

Reply:
This isn't correct. As mentioned before, we should just sync on top of the db ledger, but even if you wanted to do this, you would need to sync to a mask based directly off of the db ledger to do a transfer like this, and if that's the case you should do a commit instead of a transfer.